lib/utils/oauth/src/index.js

```diff
@@ -345,10 +345,61 @@ const getBearerToken = (authServer, id, secret, scopes, cb) => {
   });
 };
 
+const systemReadScope = 'abacus.system.read';
+
+const requiredSystemScope = {
+  system: [systemReadScope]
+};
+
+// Return an Express middleware that verifies basic authentication
+const basicStrategy = (authServer) => {
+  return (req, res, next) => {
+    debug('Authenticating request using Basic Strategy');
+
+    if (req.headers && req.headers.authorization &&
+      /^basic /i.test(req.headers.authorization)) {
+      const basicToken = req.headers.authorization;
+      let credentials;
+      try {
+        credentials = decodeBasicToken(basicToken);
+        getBearerToken(authServer, credentials[0], credentials[1],
+          systemReadScope, (err, token) => {
+            if(err) {
+              edebug('Error while retrieving bearer token due to %o', err);
+              res.status(500).end();
+              return;
+            }
+            // FIXME need authorize? we already supply scopes to getBearer?
```

Comment:
We don't really need to authorize the token, since we got it with the corresponding scopes. 
Just validate it instead.